src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -0,0 +1,410 @@
+            }
+            catch (Exception ex)
+            {
+                EqtTrace.Error("HtmlLogger : Failed to populate html file. Exception : {0}",
+                    ex.ToString());
+                ConsoleOutput.Instance.Error(false, string.Concat(HtmlResource.HtmlLoggerError), ex.Message);
+                return;
+            }
+            finally
+            {
+                if (XmlFilePath != null)
+                {
+                    this.fileHelper.Delete(XmlFilePath);
+                }
+            }
+
+            var htmlFilePathMessage = string.Format(CultureInfo.CurrentCulture, HtmlResource.HtmlFilePath, HtmlFilePath);
+            EqtTrace.Info(htmlFilePathMessage);
+            ConsoleOutput.Instance.Information(false, htmlFilePathMessage);
+        }
+
+        private string GetFilePath(string fileExtension, string fileName)
+        {
+            var fullFileFormat = $".{fileExtension}";
+            return Path.Combine(TestResultsDirPath, string.Concat("TestResult_", fileName, fullFileFormat));
+        }
+
+        private string FormatDateTimeForRunName(DateTime timeStamp)
+        {
+            return timeStamp.ToString("yyyyMMdd_HHmmss", DateTimeFormatInfo.InvariantInfo);
+        }
+
+        /// <summary>
+        /// Gives the parent execution id of a TestResult.
+        /// </summary>
+        /// <param name="testResult"></param>
+        /// <returns></returns>
+        private Guid GetParentExecutionId(TestPlatform.ObjectModel.TestResult testResult)
+        {
+            var parentExecutionIdProperty = testResult.Properties.FirstOrDefault(property =>
+                property.Id.Equals(HtmlLoggerConstants.ParentExecutionIdPropertyIdentifier));
+            return parentExecutionIdProperty == null
+                ? Guid.Empty
+                : testResult.GetPropertyValue(parentExecutionIdProperty, Guid.Empty);
+        }
+
+        /// <summary>
+        /// Gives the execution id of a TestResult.
+        /// </summary>
+        /// <param name="testResult"></param>
+        /// <returns></returns>
+        private Guid GetExecutionId(TestPlatform.ObjectModel.TestResult testResult)
+        {
+            var executionIdProperty = testResult.Properties.FirstOrDefault(property =>
+                property.Id.Equals(HtmlLoggerConstants.ExecutionIdPropertyIdentifier));
+            var executionId = Guid.Empty;
+
+            if (executionIdProperty != null)
+            {
+                executionId = testResult.GetPropertyValue(executionIdProperty, Guid.Empty);
+            }
+
+            return executionId.Equals(Guid.Empty) ? Guid.NewGuid() : executionId;
+        }
+
+        /// <summary>
+        /// Converts the time span format to readable string.
+        /// </summary>
+        /// <param name="duration"></param>
+        /// <returns></returns>
+        internal string GetFormattedDurationString(TimeSpan duration)
```

Comment:
>GetFormattedDurationString [](start = 24, length = 26)

make it private

Reply:
i want to test the GetFormattedDurationString function.